snapcraft/commands/remote.py

```diff
@@ -277,10 +279,12 @@ def _run_new_remote_build(self) -> None:
         emit.progress("Starting build")
         remote_builder.start_build()
         emit.progress("Building")
-        remote_builder.monitor_build()
-        emit.progress("Cleaning")
-        remote_builder.clean_build()
-        emit.progress("Build completed", permanent=True)
+        try:
+            remote_builder.monitor_build()
+        finally:
+            emit.progress("Cleaning")
+            remote_builder.clean_build()
+            emit.progress("Build completed", permanent=True)
```

Comment:
This will say the build completed even when it raises an exception. I'm guessing the correct behaviour is closer to:

```suggestion
        try:
            remote_builder.monitor_build()
        except Exception:
            emit.progress("Build failed", permanent=True)
            raise
        finally:
            emit.progress("Cleaning")
            remote_builder.clean_build()
        emit.progress("Build completed", permanent=True)
```

Reply:
Shows "build completed" even when it raises an exception is what I want. The `monitor_build()` included all sub building task's results that some of them may fail. Like `amd64` success but `arm64` failed. But the build task is completed.

Reply:
Change it to `build task(s) completed` maybe more clearer